src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -1208,82 +1122,126 @@ public static string GetPSModulePath(PSModulePathScope scope)
         {
-            string personalModulePath = GetPersonalModulePath();
-            string sharedModulePath = GetSharedModulePath(); // aka <Program Files> location
-            string psHomeModulePath = GetPSHomeModulePath(); // $PSHome\Modules location
+            string? personalModulePathToUse = string.IsNullOrEmpty(hkcuUserModulePath) ? GetPersonalModulePath() : hkcuUserModulePath;
+            string? sharedModulePath = GetSharedModulePath(); // aka <Program Files> location
+            string? systemModulePathToUse = string.IsNullOrEmpty(hklmMachineModulePath) ? GetPSHomeModulePath() : hklmMachineModulePath;
 
-            // If the variable isn't set, then set it to the default value
-            if (currentProcessModulePath == null)  // EVT.Process does Not exist - really corner case
+            List<string> finalModulePath = new List<string>();
+            AddPath(finalModulePath, personalModulePathToUse);
+            AddPath(finalModulePath, sharedModulePath);
+            AddPath(finalModulePath, systemModulePathToUse);
+
+            // If the existing module path if provided, we add it to the end of the list.
+            if (currentProcessModulePath != null)
             {
-                // Handle the default case...
-                if (string.IsNullOrEmpty(hkcuUserModulePath)) // EVT.User does Not exist -> set to <SpecialFolder.MyDocuments> location
-                {
-                    currentProcessModulePath = personalModulePath; // = SpecialFolder.MyDocuments + Utils.ProductNameForDirectory + Utils.ModuleDirectory
-                }
-                else // EVT.User exists -> set to EVT.User
-                {
-                    currentProcessModulePath = hkcuUserModulePath; // = EVT.User
-                }
+                AddPath(finalModulePath, currentProcessModulePath);
+            }
+
+            return string.Join(Path.PathSeparator, finalModulePath);
+        }
 
-                if (string.IsNullOrEmpty(currentProcessModulePath))
+        /// <summary>
+        /// Adds the path if not present to the final module path list. Each
+        /// path is split by the path separator and added to the final list.
+        /// </summary>
+        /// <param name="finalModulePath">The final module path list.</param>
+        /// <param name="path">The path to add.</param>
+        private static void AddPath(List<string> finalModulePath, ReadOnlySpan<char> path)
+        {
+            ReadOnlySpan<char> dirSeparatorChars = stackalloc char[] {
+                Path.DirectorySeparatorChar,
+                Path.AltDirectorySeparatorChar
+            };
+
+#if UNIX
+            IEqualityComparer<char>? valueComparer = null;
+#else
+            IEqualityComparer<char> valueComparer = new WindowsPathEnvComparer();
+#endif
+
+            foreach (Range segment in path.Split(Path.PathSeparator))
```

Comment:
Should we split by alternative path separator on Windows too?

Reply:
Is there an alternative path separator on Windows. The only one I know is the directory separator char but the PATH separators is always the one char for all platforms.